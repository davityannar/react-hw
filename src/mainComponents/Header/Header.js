import React, {Component} from 'react';
import {  withRouter, Link,Route } from 'react-router-dom'
import {connect} from 'react-redux'
import  Search  from '../Search/Search';
import OneFilm from "../oneFilm/oneFilm"
import SearchButton from '../Search/SearchButton/SearchButton'
import './Header.scss';

export class Header extends Component {
    constructor(props) {
        super(props)
      }     

    render() {               
       
      let header, headerContent;
      if( this.props.location.pathname === '/' || this.props.location.pathname === '/movies' ) {
        header = <p className='netflix'>netflixroulette</p>;
        headerContent = <><p className='find-your-movie'>Find your movie</p><Search /></>;
      } else if ( this.props.match.params.id ) {
        let cardToShow = this.props.apiData.filter( card => card.id === Number(this.props.match.params.id))[0];

        header = <><p className='netflix'>netflixroulette</p><SearchButton buttonClass='backToFilter' ><Link to={{pathname: `/movies`, search: `?search=${this.props.searchValue}&searchBy=${this.props.searchBy}&sortBy=${this.props.sortBy}&sortOrder=desc&limit=20`}}>Search</Link></SearchButton></>;
        headerContent = <OneFilm 
          cardImgPath={cardToShow.poster_path} 
          cardTitle={cardToShow.title}
          cardGenres={cardToShow.genres.join(', ')}
          cardReleaseDate={cardToShow.release_date.slice(0, 4)}
          cardRuntime={cardToShow.runtime}
          cardFilmOverview={cardToShow.overview} />
      }
            
            return (<div className='header-image'>
            <div className='header'>
            <div className='logo-header'>
            {header}
            </div>
            {headerContent}
            </div>
            </div>)      
            
    }
}
const mapStateToProps = (state) => ({apiData: state.search.apiData, sortBy: state.search.sortBy, searchValue: state.search.searchValue, searchBy: state.search.searchBy})

export default withRouter(connect(mapStateToProps)(Header))
