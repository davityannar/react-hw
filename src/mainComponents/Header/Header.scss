
.header {
    display: flex;
    flex-direction: column;
    max-width: 1024px;
    margin: 0 auto;
  }

.header-image {
    background: linear-gradient(
      rgba(0, 0, 0, 0.9), 
      rgba(0, 0, 0, 0.5)
    ), url('../../assets/headerBackground.jpg');
    background-size: cover;
    background-repeat: no-repeat;
  }

  .netflix {
    margin: 25px 0;
    padding-top: 50px;
    color: #f55263;
}
 
.find-your-movie {
  margin: 20px 0;
  text-transform: uppercase;
  color: #fff;
  font-weight: bold;
}