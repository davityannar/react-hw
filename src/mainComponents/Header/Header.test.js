import React from 'react'
import Header from './Header'

describe('Tests Header container', () => {
  it('Matching snapshot', () => {
    const container = shallow(<Header />)

    expect(container).toMatchSnapshot();
  })
})