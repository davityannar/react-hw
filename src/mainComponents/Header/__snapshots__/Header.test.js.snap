// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Header container Matching snapshot 1`] = `
<div
  className="header-image"
>
  <div
    className="header"
  >
    <p
      className="netflix"
    >
      netflixroulette
    </p>
    <p
      className="find-your-movie"
    >
      Find your movie
    </p>
    <Search />
  </div>
</div>
`;
